syft/pkg/cataloger/wordpress/parse_plugin.go

```diff
@@ -0,0 +1,94 @@
+package wordpress
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"regexp"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+var patterns = map[string]*regexp.Regexp{
+	// match example:	"Plugin Name: WP Migration"	--->	WP Migration
+	"name": regexp.MustCompile(`(?i)plugin name:\s*(?P<name>.+)`),
+
+	// match example:	"Version: 5.3"				--->	5.3
+	"version": regexp.MustCompile(`(?i)version:\s*(?P<version>[\d.]+)`),
+
+	// match example:	"License: GPLv3"			--->	GPLv3
+	"license": regexp.MustCompile(`(?i)license:\s*(?P<license>\w+)`),
+
+	// match example:	"Author: MonsterInsights"	--->	MonsterInsights
+	"author": regexp.MustCompile(`(?i)author:\s*(?P<author>.+)`),
+
+	// match example:	"Author URI: https://servmask.com/"	--->	https://servmask.com/
+	"author_uri": regexp.MustCompile(`(?i)author uri:\s*(?P<author_uri>.+)`),
+}
+
+type pluginData struct {
+	Licenses            []string `mapstructure:"licenses" json:"licenses,omitempty"`
+	pkg.WordpressPlugin `mapstructure:",squash" json:",inline"`
+}
+
+func parseWordpressPluginFiles(_ file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	var pkgs []pkg.Package
+	var fields = make(map[string]interface{})
+
+	bytes, err := io.ReadAll(reader)
```

Comment:
we're really only expecting to parse the top comment of the file. I think we should try a best-effort at not searching the entire file, but instead to the first `*/` or only the first 100 lines. 

That's my initial take since this is capturing on potentially several PHP files. What do you think @disc ?

Reply:
Hey, I agree. Will read first N kilobytes (for example, 4kb) into a buffer instead.